### PR TITLE
geoconda 2024 update

### PR DESCRIPTION
**Pull request by ktiits**
## Proposed changes

Added new version of geoconda to Puhti and Mahti.

https://csc-guide-preview.rahtiapp.fi/origin/ktiits-geoconda-2024/apps/geoconda/ 

## Checklist before requesting a review

- [ ] I have followed the instructions in the [Contributing](https://github.com/CSCfi/csc-user-guide/blob/master/CONTRIBUTING.md) and [Styleguide](https://github.com/CSCfi/csc-user-guide/blob/master/STYLEGUIDE.md) documents.
- [ ] My pull request passes the automated tests. If not, visit [pull requests at Travis CI](https://app.travis-ci.com/github/CSCfi/csc-user-guide/pull_requests) and have a look at the job log.
- [ ] I have included a link to the appropriate [preview page](https://csc-guide-preview.2.rahtiapp.fi/origin/) (select your branch from the list).

### CODE REVIEW

File: docs/support/wn/apps-new.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Applications
 
+## Geoconda 3.11.9, 7.8.2024
+
+[geoconda](../../apps/geoconda.md) version 3.11.9 has been added
+and set as the default module on Puhti and Mahti. Besides package updates, it includes a few new packages: `geo2ml`, `h3pandas`, `openeo`, `pcraster` and `r5py`. From packages available in previous geoconda, `copc-lib` and `osmnx` unfortunatelly could not be added this time.
```
Comment (msainio) [inline docs/support/wn/apps-new.md:6]:
```suggestion
and set as the default module on Puhti and Mahti. Besides package updates, it includes a few new packages: `geo2ml`, `h3pandas`, `openeo`, `pcraster` and `r5py`. From packages available in previous geoconda, `copc-lib` and `osmnx` unfortunately could not be added this time.
```

File: docs/apps/geoconda.md [hunk 2]
```diff
@@ -12,64 +12,69 @@ includes following python packages:
...
--   [geopy](https://geopy.readthedocs.io/) - client for several popular geocoding web services. NEW 2023
+-   [geopy](https://geopy.readthedocs.io/) - client for several popular geocoding web services.
+-   [geo2ml](https://github.com/mayrajeo/geo2ml) - for preparing spatial data for machine learning NEW 2024
+-   [h3pandas](https://h3-pandas.readthedocs.io/en/latest/) - for hexagonal geospatial indexing system, with Pandas and GeoPandas. NEW 2024
 -   **[jupyter]** - Jupyter Notebooks and JupyterLab, best to use with [Puhti web interface](../computing/webinterface/index.md) and [Jupyter](../computing/webinterface/jupyter.md)
 -   [laspy](https://pythonhosted.org/laspy/) - for reading, modifying, and creating .LAS LIDAR files. 
--   [landsatlinks](https://github.com/ernstste/landsatlinks) - download Landsat Collection 2 Level 1. NEW 2023
--   [leafmap](https://leafmap.org/) - for geospatial analysis and interactive mapping in a Jupyter environment.  NEW 2023
+-   [landsatlinks](https://github.com/ernstste/landsatlinks) - download Landsat Collection 2 Level 1.
```
Comment (msainio) [inline docs/apps/geoconda.md:30]:
```suggestion
-   [landsatlinks](https://github.com/ernstste/landsatlinks) - for creating download URLs for Landsat Collection 2 Level 1 product bundles using the USGS/EROS Machine-to-Machine API. Use `python3.10 -m landsatlinks`.
```

File: docs/apps/geoconda.md [hunk 3]
```diff
@@ -12,64 +12,69 @@ includes following python packages:
...
     `list-packages`
     
 Additionally geoconda includes:
 
 -   [spyder] - Scientific Python Development Environment with graphical interface (similar to RStudio for R). 
--   [GDAL/OGR](../apps/gdal.md) commandline tools 
+-   **[GDAL/OGR](../apps/gdal.md)** commandline tools 
 -   [GMT] The Generic Mapping Tools 
--   [landsatlinks](https://github.com/ernstste/landsatlinks) - for creating download URLs for Landsat Collection 2 Level 1 product bundles using the USGS/EROS Machine-to-Machine API. Use `python3.10 -m landsatlinks`. NEW 2023
+-   [landsatlinks](https://github.com/ernstste/landsatlinks) - for creating download URLs for Landsat Collection 2 Level 1 product bundles using the USGS/EROS Machine-to-Machine API. Use `python3.10 -m landsatlinks`.
```
Comment (msainio) [inline docs/apps/geoconda.md:77]:
```suggestion
```

File: docs/apps/geoconda.md [hunk 4]
```diff
@@ -12,64 +12,69 @@ includes following python packages:
...
 -   [PDAL] Point Data Abstraction Library
```
Comment (msainio) [inline docs/apps/geoconda.md:78]:
```suggestion
```
Already listed earlier on the page.

File: docs/apps/geoconda.md (hunk 3 above)
Comment (ktiits) [inline docs/apps/geoconda.md:77]: -   [landsatlinks](https://github.com/ernstste/landsatlinks) - for creating download URLs for Landsat Collection 2 Level 1 product bundles using the USGS/EROS Machine-to-Machine API. Use `python3.10 -m landsatlinks`.

File: docs/apps/geoconda.md (hunk 4 above)
Comment (ktiits) [reply docs/apps/geoconda.md:78]: This PDAL command-line tool, above is pdal Python library. Keeping both.

File: docs/apps/geoconda.md (hunk 3 above)
Comment (ktiits) [reply docs/apps/geoconda.md:77]: Removed from above list, keeping this.

File: docs/apps/geoconda.md (hunk 4 above)
Comment (msainio) [reply docs/apps/geoconda.md:77]: The linked page is https://github.com/PDAL/python

Comment (msainio) [reply docs/apps/geoconda.md:77]: The command-line tool is in the list as:

`[pdal](https://pdal.io/) - for lidar data`

Comment (ktiits) [reply docs/apps/geoconda.md]: Fixed PDAL links in a commit.

File: docs/apps/geoconda.md [hunk 5]
```diff
@@ -130,6 +135,12 @@ module load geoconda
 spyder
 ```
 
+### r5py memory settings
+`r5py` by default does not correctly understand how much memory it has available in a supercomputer so, it has to be defined manually. It is using Java in the background, so add environmental variable to set maximum memory available for Java: 
+
+* `export _JAVA_OPTIONS="-Xmx4g"` from command-line before starting Python OR
+* `os.environ["_JAVA_OPTIONS"] = "-Xmx4g"` in the beginning of you Python code.
```
Comment (msainio) [inline docs/apps/geoconda.md:142]:
```suggestion
* `os.environ["_JAVA_OPTIONS"] = "-Xmx4g"` in the beginning of your Python code.
```

File: docs/apps/geoconda.md [hunk 6]
```diff
@@ -12,65 +12,69 @@ includes following python packages:
...
--   [metpy](https://unidata.github.io/MetPy/latest/index.html) - reading, visualizing, and performing calculations with weather data. NEW 2022
+-   [metpy](https://unidata.github.io/MetPy/latest/index.html) - reading, visualizing, and performing calculations with weather data.
 -   [movingpandas](http://movingpandas.org) - for trajectory data
 -   [networkx] - for the creation, manipulation, and study of the structure, dynamics, and functions of complex networks. [Routing examples in CSC geocomputing Github](https://github.com/csc-training/geocomputing/tree/master/python/routing)
 -   [pyproj] - performs cartographic transformations and geodetic computations.
--   [pyogrio](https://pyogrio.readthedocs.io/en/latest/index.html) - vectorized spatial vector file format I/O using GDAL/OGR. NEW 2022
--   [open3d](http://www.open3d.org/docs/release/index.html) - for 3D data processing, NEW 2023
+-   [pyogrio](https://pyogrio.readthedocs.io/en/latest/index.html) - vectorized spatial vector file format I/O using GDAL/OGR.
+-   [openeo](https://openeo.org/) - for connectin to Earth observation cloud back-ends in a simple and unified way. NEW 2024
```
Comment (msainio) [inline docs/apps/geoconda.md:37]:
```suggestion
-   [openeo](https://openeo.org/) - for connecting to Earth observation cloud back-ends in a simple and unified way. NEW 2024
```

**Review by msainio — APPROVED** [review]
LGTM!

**Pull request merged**